core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -105,10 +102,11 @@ public DockerComposeContainer(String identifier, File... composeFiles) {
     public DockerComposeContainer(String identifier, List<File> composeFiles) {
 
         this.composeFiles = composeFiles;
+        this.parsedComposeFiles = composeFiles.stream().map(ParsedDockerComposeFile::new).collect(Collectors.toSet());
```

Comment:
Perform light parsing of the docker compose files at instantiation. We could delay this if it's felt doing this is too heavy at instantiation-time, but if we do that we just need to make sure we do it _at some point_.